projects/ComposeConstraintLayout/app/src/main/java/com/example/constraintlayout/MotionFlowDemo.kt

```diff
@@ -0,0 +1,295 @@
+    }
+
+    Column( modifier = Modifier.background(Color.White)) {
+
+        val scene1 = MotionScene("""
+            {
+                Header: {
+                  name: 'flowDemo1'
+                },
+                
+                ConstraintSets: {
+                  start: {
+                    flow1: { 
+                        width: 'match_parent',
+                        height: 'match_parent',
+                        type: 'hFlow',
+                        wrap: 'none',
+                        vGap: 20,
+                        hGap: 20,
+                        centerVertically: 'parent',
+                        centerHorizontally: 'parent',
+                        contains: ['1', '2', '3', '4'],
+                      }
+                  },
+                  
+                  end: {
+                    flow2: { 
+                        width: 'match_parent',
+                        height: 'match_parent',
+                        type: 'vFlow',
+                        wrap: 'none',
+                        vGap: 20,
+                        hGap: 20,
+                        centerVertically: 'parent',
+                        centerHorizontally: 'parent',
+                        contains: ['1', '2', '3', '4'],
+                      }
+                  }
+                },
+                
+                Transitions: {
+                  default: {
+                    from: 'start',   to: 'end',
+                  }
+                }
+            }
+            """)
+
+        MotionLayout(
+            modifier    = Modifier
+                .fillMaxWidth()
+                .height(400.dp),
+            motionScene = scene1,
+            progress   = progress.value) {
+            val numArray = arrayOf("1", "2", "3", "4")
+            for (num in numArray) {
+                Button(
+                    modifier = Modifier.layoutId(num),
+                    onClick = {},
+                ) {
+                    Text(text = num)
+                }
+            }
+
+        }
+
+        Button(onClick = { animateToEnd = !animateToEnd },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(3.dp)) {
+            Text(text = "Run")
+        }
+    }
+}
+
+@Preview(group = "motion")
+@Composable
+public fun MotionFlowDemo2() {
+    var animateToEnd by remember { mutableStateOf(false) }
+
+    val progress  = remember { Animatable(0f) }
+
+    LaunchedEffect(animateToEnd) {
+        progress.animateTo(if (animateToEnd) 1f else 0f,
+            animationSpec = tween(3000))
+    }
+
+    Column( modifier = Modifier.background(Color.White)) {
+
+        val scene1 = MotionScene("""
+            {
+                Header: {
+                  name: 'flowDemo1'
+                },
+                
+                ConstraintSets: {
+                  start: {
+                    flow1: { 
+                        width: 'match_parent',
```

Comment:
Dimensions do not have a match_parent 
'wrap' | 'parent' | 'spread' | '%32' | '16:9'